examples/sandboxer.rs

```diff
@@ -0,0 +1,154 @@
+use anyhow::bail;
+use landlock::{AccessFs, Ruleset, RulesetAttr};
+use nix::fcntl::{open, OFlag};
+use nix::sys::stat::{fstat, Mode, SFlag};
+use std::os::unix::ffi::{OsStrExt, OsStringExt};
+use std::os::unix::process::CommandExt;
+use std::process::Command;
+use std::{
+    borrow::Cow,
+    env,
+    ffi::{OsStr, OsString},
+    path::PathBuf,
+};
+
+const ENV_FS_RO_NAME: &str = "LL_FS_RO";
+const ENV_FS_RW_NAME: &str = "LL_FS_RW";
+
+const ACCESS_FS_ROUGHLY_READ: AccessFs = AccessFs::from_bits_truncate(
+    AccessFs::EXECUTE.bits() | AccessFs::READ_FILE.bits() | AccessFs::READ_DIR.bits(),
+);
+
+const ACCESS_FS_ROUGHLY_WRITE: AccessFs = AccessFs::from_bits_truncate(
+    AccessFs::WRITE_FILE.bits()
+        | AccessFs::REMOVE_DIR.bits()
+        | AccessFs::REMOVE_FILE.bits()
+        | AccessFs::MAKE_CHAR.bits()
+        | AccessFs::MAKE_DIR.bits()
+        | AccessFs::MAKE_REG.bits()
+        | AccessFs::MAKE_SOCK.bits()
+        | AccessFs::MAKE_FIFO.bits()
+        | AccessFs::MAKE_BLOCK.bits()
+        | AccessFs::MAKE_SYM.bits(),
+);
+
+const ACCESS_FILE: AccessFs = AccessFs::from_bits_truncate(
+    AccessFs::READ_FILE.bits() | AccessFs::WRITE_FILE.bits() | AccessFs::EXECUTE.bits(),
+);
+
+/// Populates a given ruleset with PathBeneath landlock rules
+///
+/// # Arguments
+///
+/// * `ruleset` - The ruleset to add the rules to. Note that due to the current
+///   API, it is also returned at the end.
+/// * `paths` - An OsString that contains the paths taht are going to be
+///   restricted. Paths are separated with ":" e.g. "/bin:/lib:/usr:/proc". In
+///   case an empty string is provided, NO restrictions are applied.
+/// * `access` - The set of restrictions to apply to each of the given paths.
+///
+fn populate_ruleset(
+    ruleset: Ruleset,
+    paths: OsString,
+    access: AccessFs,
+) -> Result<Ruleset, anyhow::Error> {
+    let mut ruleset = ruleset;
+    let mut actual_access = access.clone();
+
+    if paths.len() == 0 {
+        return Ok(ruleset);
+    }
+
+    let individual_paths: Vec<PathBuf> = paths
+        .into_vec()
+        .split(|b| *b == b':')
+        .map(|v| OsStr::from_bytes(v).to_owned().into())
+        .collect();
+
+    for path in individual_paths {
+        match open(&path, OFlag::O_PATH | OFlag::O_CLOEXEC, Mode::empty()) {
+            Err(e) => {
+                bail!("Failed to open \"{}\": {}\n", path.to_string_lossy(), e);
+            }
+            Ok(parent_fd) => match fstat(parent_fd) {
+                Ok(stat) => {
+                    if (stat.st_mode & SFlag::S_IFMT.bits()) != SFlag::S_IFDIR.bits() {
+                        actual_access.insert(ACCESS_FILE);
```

Comment:
There is an issue with the logic here. It doesn't work with files:
```
LL_FS_RO="/bin:/lib:/usr:/proc:/etc:/dev/urandom" LL_FS_RW="/dev/null:/dev/full:/dev/zero:/dev/pts:/tmp" target/debug/examples/sandboxer bash -i
file: "/dev/urandom"
Error: Failed to update ruleset with "/dev/urandom": Invalid argument (os error 22)

```
An extra carriage return is also appended at the end of an error message.